src/Serializer/AbstractItemNormalizer.php

```diff
@@ -97,9 +100,21 @@ public function normalize($object, $format = null, array $context = [])
             $context['resources'][$resource] = $resource;
         }
 
+        if ($this->isAlwaysIdentifierItem($context)) {
```

Comment:
Can you inline this one? We try to reduce the number of function calls in the serialization process (which is recursive) because a function call is costly:

```php
if (($context[self::ALWAYS_IDENTIFIER_KEY] ?? false) && ($context[self::ALWAYS_IDENTIFIER_KEY] ?? false))
```

Same for `isAlwaysIdentifierProperty`, I would drop these methods.